pkg/sdk/identifier_parsers_test.go

```diff
@@ -250,3 +271,31 @@ func Test_ParseObjectIdentifierString(t *testing.T) {
 		})
 	}
 }
+
+func Test_ParseFunctionArgumentsFromString(t *testing.T) {
+	testCases := []struct {
+		Arguments string
+		Expected  []DataType
+		Error     string
+	}{
+		{Arguments: `()`, Expected: []DataType{}},
+		{Arguments: `(FLOAT, NUMBER, TIME)`, Expected: []DataType{DataTypeFloat, DataTypeNumber, DataTypeTime}},
+		{Arguments: `FLOAT, NUMBER, TIME`, Expected: []DataType{DataTypeFloat, DataTypeNumber, DataTypeTime}},
+		{Arguments: `(FLOAT, NUMBER, VECTOR(FLOAT, 20))`, Expected: []DataType{DataTypeFloat, DataTypeNumber, DataType("VECTOR(FLOAT, 20)")}},
+		{Arguments: `FLOAT, NUMBER, VECTOR(FLOAT, 20)`, Expected: []DataType{DataTypeFloat, DataTypeNumber, DataType("VECTOR(FLOAT, 20)")}},
+		{Arguments: `(VECTOR(FLOAT, 10), NUMBER, VECTOR(FLOAT, 20))`, Expected: []DataType{DataType("VECTOR(FLOAT, 10)"), DataTypeNumber, DataType("VECTOR(FLOAT, 20)")}},
+		{Arguments: `VECTOR(FLOAT, 10)| NUMBER, VECTOR(FLOAT, 20)`, Error: "expected a comma delimited string but found |"},
```

Comment:
please add a case with incorrect VECTOR, like VECTOR(VARCHAR, 5), and/or VECTOR(5, FLOAT)

Reply:
Added additional validation for vectors and added test cases for them